pkg/trace/roundtripper_test.go

```diff
@@ -320,22 +203,6 @@ func otelCallRoundTripper(t *testing.T, state *roundtripperState) []map[string]i
 	return state.Ended()
 }
 
-func honeycombInitRoundTripperState(t *testing.T) *roundtripperState {
-	recorder := tracetest.NewSpanRecorder()
-	server := httptest.NewServer(
-		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			prop, err := propagation.UnmarshalHoneycombTraceContext(r.Header.Get(propagation.TracePropagationHTTPHeader))
```

Comment:
I'm not sure why anyone would write an HTTP handler that digs directly into trace headers like this.  That's the framework's job, right?

After I ripped out the support for Honeycomb headers this started to fail pretty badly, and for good reason.  The `"X-Honeycomb-Trace"` header had disappeared.

I'm going to assume this is a test-only issue.  It should be only framework-y things that look at these headers.